### PR TITLE
Add normalize by bin width option for sliceviewer

### LINKED ISSUE #27359: Sliceviewer: bin width normalisation
Worth deciding what should be done to solve this: Potentially a toggle option for Normalisation by None or Bin Width as in normal 1D plots in Workbench?

### Expected behavior  

In Mantidplot Spectrumviewer seemd to normalise its lineplots by BinWidth, so I'd expect the lineplots in SliceViewer in Workbench to do the same.

### Actual behavior

WB Sliceviewer lineplots NOT normlaised by BinWidth

### Steps to reproduce the behavior

**In MantidPlot:**
Load [MAR11060](https://github.com/mantidproject/mantid/files/3836495/MAR11060.zip) from TrainingCourseData
Load SpectrumViewer for this data and observe how the Time-of-flight subplot (lower lineplot) has one large peak (~1500) and a smaller one (~6000).
PlotSpectrumView 

**In Workbench**
Load [MAR11060](https://github.com/mantidproject/mantid/files/3836495/MAR11060.zip) from TrainingCourseData
Load SpectrumViewer for this data and observe how the Time-of-flight subplot (lower lineplot) has one SMALLER peak (~1500) and a LARGER one (~6000). Opposite to Plot!!!
WBSliceView

Convince yourself that the difference is that SliceViewer in WB is NOT normalising by BinWidth by plotting a 1D spectrum of this MAR11060 workspace (easier to do this in WB) and then right-click on the plot and change the Normalisation. None gives whats seen in SliceViewer in WB and Normalise by BinWidth gives whats seen in SpectrumViewer in Plot.
NormbyBinWidthvsNone

### Platforms affected
Workbench most recent nightly and v4.1

### PR DESCRIPTION
**Pull request by alicerussell1**
~~**This PR cannot be merged until #27468**~~

**Description of work.**
This PR adds the option to normalize by bin width for matrix workspaces in slice viewer. In mantidplot this was done in the spectrum viewer by rebinning the data. This instead normalises the data by dividing it by the bin width.

**To test:**
See issue for detailed example.

Fixes #27359 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by Harrietbrown — CHANGES_REQUESTED** [review]
The curves are able to be normalized by bin width and it all works as expected. Some minor changes. 

I have noticed that the range on the curves do not autoscale, because of this and the monitor curve being so much larger than any other, you need to manualy scale the plot to be able to see them. Is there an issue for this already?

**Comment by alicerussell1** [comment]
I've rebased this onto master as it included some of the commits from the earlier PR that has now been merged. For the autoscaling there is a tick box which if selected it should do the autoscaling and when you normalise it should change the scale.

File: qt/applications/workbench/workbench/plotting/figureinteraction.py [hunk 1]
```diff
@@ -564,25 +586,30 @@ def _toggle_normalization(self, ax):
...
+            ax.relim()
         ax.autoscale()
         self.canvas.draw()
 
     def _can_toggle_normalization(self, ax):
         """
-        Return True if no plotted workspaces are distributions and all curves
-        on the figure are either distributions or non-distributions. Return
+        Return True if no plotted workspaces are distributions, all curves
+        on the figure are either distributions or non-distributions
```
[the remark this answers is not in the document]
Comment (alicerussell1) [reply qt/applications/workbench/workbench/plotting/figureinteraction.py]: This has been changed in the earlier PR that this was based off so it should be correct now I've rebased it.

File: qt/python/mantidqt/widgets/sliceviewer/test/test_sliceviewer_model.py [hunk 2]
```diff
@@ -160,6 +160,27 @@ def test_model_matrix(self):
...
+
+    def test_matrix_workspace_cannot_be_normalized_if_a_distribution(self):
+        model = SliceViewerModel(self.ws2d_histo)
+        self.assertFalse(model.can_normalize_workspace())
+
+    def test_3d_workspaces_cannot_be_normalized(self):
+        model = SliceViewerModel(self.ws_MD_3D)
+        self.assertFalse(model.can_normalize_workspace())
+
+        model = SliceViewerModel(self.ws_MDE_3D)
```
[the remark this answers is not in the document]
Comment (alicerussell1) [reply qt/python/mantidqt/widgets/sliceviewer/test/test_sliceviewer_model.py]: They are two different workspaces but I intended to do them as separate tests so I've changed it

**Comment by Harrietbrown** [comment]
It does change the scale when rebinning, the issue I was talking about rescaling the curves when moving the mouse over the colourplot like it does in sliceviewer in Mantidplot.

**Review by Harrietbrown — APPROVED** [review]
The issue with the curve not rescaling is unrelated to this PR so no reason to hold it back for that. Good work.

**Pull request merged**